File: saude_ruby/lib/saude_ruby/aluno.rb
# lib/saude_ruby/aluno.rb
module SaudeRuby
  class Aluno
    attr_reader :peso, :altura, :idade, :sexo, :cintura, :quadril

    def initialize(peso: nil, altura: nil, idade: nil, sexo: nil, cintura: nil, quadril: nil)
      @peso = peso
      @altura = altura
      @idade = idade
      @sexo = sexo
      @cintura = cintura
      @quadril = quadril
    end

    def imc
      IMC.calcular(@peso, @altura)
    end

    def tmb
      TMB.calcular(@peso, @altura * 100, @idade, @sexo) # altura convertida para cm
    end

    def rcq
      RCQ.calcular(@cintura, @quadril)
    end

    def pg
      PG.calcular(imc, @idade, @sexo)
    end
  end
end

File: saude_javascript/lib/aluno.js
const IMC = require('./imc');
const TMB = require('./tmb');
const RCQ = require('./rcq');
const PG = require('./pg');

class Aluno {
    constructor({ peso = null, altura = null, idade = null, sexo = null, cintura = null, quadril = null } = {}) {
        this.peso = peso;
        this.altura = altura;
        this.idade = idade;
        this.sexo = sexo;
        this.cintura = cintura;
        this.quadril = quadril;
    }

    imc() {
        return IMC.calcular(this.peso, this.altura);
    }

    tmb() {
        // altura convertida para cm (por isso multiplicada por 100)
        return TMB.calcular(this.peso, this.altura * 100, this.idade, this.sexo);
    }

    rcq() {
        return RCQ.calcular(this.cintura, this.quadril);
    }

    pg() {
        return PG.calcular(this.imc(), this.idade, this.sexo);
    }
}

module.exports = Aluno;
